src/Web/Scotty/Action.hs

```diff
@@ -264,12 +269,14 @@ instance Parsable () where
 instance (Parsable a) => Parsable [a] where parseParam = parseParamList
 
 instance Parsable Bool where
-    parseParam t = if t' == T.toCaseFold "true"
-                   then Right True
-                   else if t' == T.toCaseFold "false"
-                        then Right False
-                        else Left "parseParam Bool: no parse"
-        where t' = T.toCaseFold t
+    parseParam t
+        | t' == true = Right True
+        | t' == false = Right False
+        | otherwise = Left "parseParam Bool: no parse"
+        where
+            t' = T.toCaseFold t
+            true = T.toCaseFold "true"
+            false = T.toCaseFold "false"
```

Comment:
I cleaned up this block by changing nested if/then to guards, and factored out `true` and `false` so that a sufficiently-advanced compiler could float them if it wanted.